lib/dialects/postgres/query-generator.js

```diff
@@ -805,8 +806,10 @@ var QueryGenerator = {
   },
 
   pgEnumAdd: function(tableName, attr, value, options) {
-    var enumName = this.pgEnumName(tableName, attr)
-      , sql = 'ALTER TYPE ' + enumName + ' ADD VALUE ' + this.escape(value);
+    var enumName = this.pgEnumName(tableName, attr);
+    var sql = semver.gte(this.sequelize.options.databaseVersion, '9.3.0')
+        ? 'ALTER TYPE ' + enumName + ' ADD VALUE IF NOT EXISTS ' + this.escape(value)
+        : 'ALTER TYPE ' + enumName + ' ADD VALUE ' + this.escape(value);
```

Comment:
I totally agree! I was thinking of doing something like this but thought it might have hurt readability a bit, so decided against it. Let me fix it and squash these commits real quick.